schema.sql

```diff
@@ -32,16 +34,27 @@ CREATE TABLE vets(
 );
 
 CREATE TABLE specializations(
-  id INT GENERATED ALWAYS AS IDENTITY,
-  species_id INT REFERENCES species(id),
-  vet_id INT REFERENCES vets(id),
-  PRIMARY KEY(id)
+  species_id  INT,
+  vets_id     INT,
+  FOREIGN KEY (species_id) REFERENCES species (id),
+  FOREIGN KEY (vets_id) REFERENCES vets (id),
+  PRIMARY KEY (species_id, vets_id)
 );
 
 CREATE TABLE visits(
-  id INT GENERATED ALWAYS AS IDENTITY,
-  animal_id INT REFERENCES animals(id),
-  vet_id INT REFERENCES vets(id),
+  animal_id  INT,
+  vet_id     INT,
   date_of_visit DATE,
-  PRIMARY KEY(id)
-);
\ No newline at end of file
+  id INT GENERATED ALWAYS AS IDENTITY,
+  FOREIGN KEY (animal_id) REFERENCES animals (id),
+  FOREIGN KEY (vet_id) REFERENCES vets (id),
+  PRIMARY KEY (id)
+);
+
+ALTER TABLE owners ADD COLUMN email VARCHAR(120);
+
+CREATE INDEX animal_id_idx ON visits (animal_id);
+
+CREATE INDEX vet_id_idx ON visits (vet_id);
+
+CREATE INDEX owners_email_idx ON owners (email);
```

Comment:
## Optional
- [ ] I suggest making the index on the case insensitive version of the email, 
As email addresses are not case sensitive, we might need to look for the emails in a case insensitive way. There is an option to reflect that in the index by using:
`CREATE INDEX ON owners ((lower(email)));`

Reply:
Thanks for this suggestion, trying it out 👍🏼.